core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -42,7 +38,7 @@ boolean allocate(Slice s, int size) {
             allocated.getAndAdd(-size);
             throw new OakOutOfMemoryException(String.format("Block %d is out of memory", id));
         }
-        s.associateBlockAllocation(id, now, size, buffer);
+        s.associateBlockAllocation(id, now, size, memAddress, capacity);
```

Comment:
I also wonder, why to pass the block's capacity all over...? All blocks have the same capacity used to `UnsafeUtils.unsafe.allocateMemory(this.capacity)` in every Block creation. Can we have (for example) static  `getCapacity()` interface in Block class or MemoryManager class/interface? And this can be used anywhere when needed...

Reply:
I needed to attach the capacity with slice since, when wrapping the address with bytebuffer, I just have access to slice.
I can make the capacity static in the Native allocator but wont that affect another Oak instance that wants different capacity?

Reply:
We currently have one BlockPool for all Oak instances, meaning only one block capacity supported. It can be differently defined when the first Oak instance (on the same JVM) is built, after that block capacity cannot be changed.

Reply:
changed to use the slice length in order to wrap the bytebuffer